### PR TITLE
Valgrind tests

### PR DESCRIPTION
**Pull request by bscothern**
This should be reviewed after [PR13](https://github.com/HollowCore/HollowCore/pull/13).

It adds a valgrind version of the tests to `travis-ci` which exits with an error code of 1 if there are any memory issues in the library. This was added to properly validate efforts to have system threads that can be cleaned up properly.

There are other parts of the library that have issues that I haven't had a chance to fix yet.

### CODE REVIEW

File: Source/Container/HCMap_Internal.h [hunk 1]
```diff
@@ -51,4 +52,9 @@ void HCMapPairDestroy(HCMapPairRef self);
 HCBoolean HCMapPairIsEqual(HCMapPairRef self, HCMapPairRef other);
 HCInteger HCMapPairHashValue(HCMapPairRef self);
 
+//----------------------------------------------------------------------------------------------------------------------------------
+// MARK: - Iteration
+//----------------------------------------------------------------------------------------------------------------------------------
+HCInteger _HCMapIteratorMinimumSizeRequiredForState(void);
```
Comment (mattstoker) [inline Source/Container/HCMap_Internal.h:58]: Does this have to be public for external access? Or is it just for testing that it is public? If it is just for testing, we can move it to the _Internal header and include it in the test case.

Comment (mattstoker) [reply Source/Container/HCMap_Internal.h:58]: Oh! This is already the internal header. Nice.

File: Test/HCThread_Internal.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//
+//  HCThread_Internal.c
```
Comment (mattstoker) [inline Test/HCThread_Internal.c:2]: _Internal.c? We don't ever need an internal implementation file, do we? This should just be in HCThread.c

**Review by mattstoker — APPROVED** [review]

**Comment by mattstoker** [comment]
One comment, but the Travis tests are going to be awesome! I'll check out a few of them as well. The failures still manifest when running valgrind locally, right?

**Comment by mattstoker** [comment]
Also, can you modify README.md to show how to run valgrind locally?

**Comment by mattstoker** [comment]
Should we merge this in, or wait until the memory tests pass?

**Comment by mattstoker** [comment]
I'm good either way. The previous merged version would not have succeeded either.

File: Test/HCThread_Internal.c (hunk 2 above)
Comment (bscothern) [reply Test/HCThread_Internal.c:2]: I am testing implementation details and internal aspects is why I ended up doing it this way. I can move it if you want.

**Comment by mattstoker** [comment]
It all passes now with no memory leaks!!!

**Pull request merged**